resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesVolumeUtils.scala

```diff
@@ -0,0 +1,142 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s
+
+import scala.collection.mutable.HashMap
+
+import io.fabric8.kubernetes.api.model._
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.k8s.Config._
+
+private[spark] object KubernetesVolumeUtils {
+
+  /**
+   * Given hostPath volume specs, add volume to pod and volume mount to container.
+   *
+   * @param pod original specification of the pod
+   * @param container original specification of the container
+   * @param sparkConf Spark configuration
+   * @param prefix the prefix for volume configuration
+   * @return a tuple of (pod with the volume(s) added, container with mount(s) added)
+   */
+  def addVolumes(
+      pod: Pod,
+      container: Container,
+      sparkConf: SparkConf,
+      prefix : String): (Pod, Container) = {
+    val hostPathVolumeSpecs = parseHostPathVolumesWithPrefix(sparkConf, prefix)
+    addHostPathVolumes(pod, container, hostPathVolumeSpecs)
+  }
+
+  /**
+   * Extract Spark volume configuration properties with a given name prefix.
+   *
+   * @param sparkConf Spark configuration
+   * @param prefix the given property name prefix
+   * @param volumeTypeKey the given property name prefix
+   * @return a Map storing with volume name as key and spec as value
+   */
+  def parseVolumesWithPrefix(
+      sparkConf: SparkConf,
+      prefix: String,
+      volumeTypeKey: String): Map[String, KubernetesVolumeSpec] = {
+    val volumes = HashMap[String, KubernetesVolumeSpec]()
+    val properties = sparkConf.getAllWithPrefix(s"$prefix$volumeTypeKey.").toList
+    // Extract volume names
+    properties.foreach {
+      k =>
+        val keys = k._1.split("\\.")
+        if (keys.nonEmpty && !volumes.contains(keys(0))) {
+          volumes.update(keys(0), KubernetesVolumeSpec.emptySpec())
+        }
+    }
+    // Populate spec
+    volumes.foreach {
+      case (name, spec) =>
+        properties.foreach {
+          k =>
+            k._1.split("\\.") match {
+              case Array(`name`, KUBERNETES_VOLUMES_MOUNT_KEY, KUBERNETES_VOLUMES_PATH_KEY) =>
+                spec.mountPath = Some(k._2)
+              case Array(`name`, KUBERNETES_VOLUMES_MOUNT_KEY, KUBERNETES_VOLUMES_READONLY_KEY) =>
+                spec.mountReadOnly = Some(k._2.toBoolean)
+              case Array(`name`, KUBERNETES_VOLUMES_OPTIONS_KEY, option) =>
+                spec.optionsSpec.update(option, k._2)
+              case _ =>
+                None
+            }
+        }
+    }
+    volumes.toMap
+  }
+
+  /**
+   * Extract Spark hostPath volume configuration properties with a given name prefix and
+   * return the result as a Map.
+   *
+   * @param sparkConf Spark configuration
+   * @param prefix the given property name prefix
+   * @return a Map storing with volume name as key and spec as value
+   */
+  def parseHostPathVolumesWithPrefix(
```

Comment:
Looks like you don't really need this function as it's just a wrapper of `parseVolumesWithPrefix `.